csharp/test/TestCommon/Collections.cs

```diff
@@ -55,15 +55,11 @@ public static bool Equals<T>(T[]? arr1, T[]? arr2)
             return true;
         }
 
-        public static bool Equals<T>(IReadOnlyList<T>? lhs, IReadOnlyList<T>? rhs)
-        {
-            return Equals<T>(lhs as IReadOnlyCollection<T>, rhs as IReadOnlyCollection<T>);
-        }
+        public static bool Equals<T>(IReadOnlyList<T>? lhs, IReadOnlyList<T>? rhs) =>
```

Comment:
These should be removed. We should just use Linq's Enumerable.SequenceEqual.

Reply:
I will do that on a separate PR, lot of changes already